src/tests/ftest/control/dmg_network_scan.py

```diff
@@ -0,0 +1,187 @@
+
+        numas_devs = {}
+        for i, numa in enumerate(numas):
+            net_dev = []
+            split_numa = numa.split("\n")
+            for line in split_numa:
+                if "Net" in line or "OpenFabrics" in line:
+                    # Get the devices i.e. eth0, ib0, ib1
+                    net_dev.append("".join(line.strip().split('"')[1::2]))
+            numas_devs[i] = net_dev
+
+        return numas_devs
+
+    def get_dev_provider(self, device):
+        """Get expected values of provider with fi_info."""
+        fi_info = os.path.join(
+            self.bin, "fi_info -d {} | grep provider | sort -u".format(device))
+        kwargs = {
+            "cmd": fi_info,
+            "allow_output_check": "combined",
+            "verbose": False,
+            "shell": True,
+        }
+        try:
+            output = process.run(**kwargs)
+        except process.CmdError as error:
+            # Command failed or possibly timed out
+            msg = "Error occurred running '{}': {}".format(fi_info, error)
+            self.fail(msg)
+
+        # Clean up string
+        provider = "none"
+        if (output.stdout != ""):
+            provider = output.stdout.splitlines()[0].split()[-1]
+
+        return provider
+
+    def test_dmg_network_scan_basic(self):
+        """
+        JIRA ID: DAOS-2516
+        Test Description: Test basic dmg functionality to scan the network
+        devices on the system.
+        :avocado: tags=all,tiny,pr,hw,dmg,network_scan,basic
+        """
+        # Create dmg command
+        dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        dmg.get_params(self)
+
+        # Update hostlist value for dmg command
+        port = self.params.get("port", "/run/server_config/*")
+        servers_with_ports = [
+            "{}:{}".format(host, port) for host in self.hostlist_servers]
+        dmg.hostlist.update(",".join(servers_with_ports), "dmg.hostlist")
+
+        try:
+            dmg_out = dmg.run()
+        except process.CmdError as details:
+            self.fail("dmg command failed: {}".format(details))
+
+        # This parse asumes that the device information is listed on separate
+        # lines and with this format 'info_type: info'
+        if isinstance(dmg_out.stdout, str):
+            pout = {}
+            net_items = ["numa_node:", "iface:", "provider:"]
+            for i, line in enumerate(dmg_out.stdout.splitlines()):
+                for j, item in enumerate(net_items):
+                    if item in line:
+                        pout[i + j] = line.strip('\t')
+
+        # Get info from tools
+        exp_out = []
+        exp_devs = self.get_net_info("sockets") + self.get_net_info("psm2")
+        t_hfi = {"hfi1_0": "ib0", "hfi1_1": "ib1"}
+        for numa, devs in self.get_numa_info().items():
+            if devs:
+                n_devs = [dev for dev in devs if dev in exp_devs]
+                n_devs.append("lo") if numa == 0 else None
```

Comment:
(pylint-expression-not-assigned) Expression "n_devs.append('lo') if numa == 0 else None" is assigned to nothing